microbosh-setup/azuredeploy.json

```diff
@@ -0,0 +1,262 @@
+      "location": "[parameters('location')]",
+      "dependsOn": [
+        "[concat('Microsoft.Storage/storageAccounts/', parameters('newStorageAccountName'))]",
+        "[concat('Microsoft.Network/networkInterfaces/', variables('nicName'))]"
+      ],
+      "properties": {
+        "hardwareProfile": {
+          "vmSize": "[parameters('vmSize')]"
+        },
+        "osProfile": {
+          "computername": "[parameters('vmName')]",
+          "adminUsername": "[parameters('adminUsername')]",
+          "adminPassword": "[parameters('adminPassword')]"
+        },
+        "storageProfile": {
+          "imageReference": {
+                        "publisher": "[variables('imagePublisher')]",
+                        "offer": "[variables('imageOffer')]",
+                        "sku" : "[variables('ubuntuOSVersion')]",
+                        "version":"latest"
+                    },
+          "osDisk" : {
+              "name": "osdisk",
+              "vhd": {
+                  "uri": "[concat('http://',parameters('newStorageAccountName'),'.blob.core.windows.net/',variables('vmStorageAccountContainerName'),'/',parameters('vmName'),'-osdisk.vhd')]"
+              },
+              "caching": "ReadWrite",
+              "createOption": "FromImage"
+          },
+          "dataDisks": [
+            {
+              "name": "datadisk1",
+              "diskSizeGB": "30",
+              "lun": 0,
+              "createOption": "empty",
+              "vhd": { "uri": "[concat('http://',parameters('newStorageAccountName'),'.blob.core.windows.net/',variables('vmStorageAccountContainerName'),'/',parameters('vmName'),'datadisk1.vhd')]" }
+            }
+          ]
+        },
+        "networkProfile": {
+          "networkInterfaces": [
+            {
+              "id": "[resourceId('Microsoft.Network/networkInterfaces',variables('nicName'))]"
+            }
+          ]
+        }
+      }
+    },
+    {
+      "type": "Microsoft.Compute/virtualMachines/extensions",
+      "name": "[concat(parameters('vmName'),'/customscript')]",
+      "apiVersion": "[variables('api-version')]",
+      "location": "[parameters('location')]",
+      "dependsOn": ["[concat('Microsoft.Compute/virtualMachines/', parameters('vmName'))]"
+      ],
+      "properties": {
+        "publisher": "Microsoft.OSTCExtensions",
+        "type": "CustomScriptForLinux",
+        "typeHandlerVersion": "1.2",
+        "settings": {
+          "fileUris": [
+            "[concat('https://raw.githubusercontent.com/', variables('branch'),'/azure-quickstart-templates/master/microbosh-setup/launch_script.sh')]"
+          ],          
+           "commandToExecute":  "[concat('sh launch_script.sh',' https://raw.githubusercontent.com/', variables('branch'),'/azure-quickstart-templates/master/microbosh-setup/')]",
+          "newStorageAccountName:":"[parameters('newStorageAccountName')]",
+          "virtualNetworkName":"[variables('virtualNetworkName')]",
```

Comment:
I guess the question is why are you passing these into the extension. Does WALA use this information in the logs for better debugging?